### PR TITLE
Category common button Add

### LINKED ISSUE #992: ButtonCategoryAdd
- [x] Create common survey component 
- [x] When click, do the post and then invoke the callback 
- [x] test it in nodeDefDetails and List view
- [x] Remove categories redux actions and reducer if they still exists

### PR DESCRIPTION
**Pull request by ramirobg94**
resolves #992

### CODE REVIEW

File: webapp/components/survey/CategoryDetails/store/actions/category/useInit.js [hunk 1]
```diff
@@ -1,54 +1,28 @@
...
 
-    if (A.isEmpty(categoryUuid)) {
-      category = await _createCategory({ surveyId, onCategoryCreated, dispatch })
-    } else {
-      category = await _fetchCategory({ surveyId, categoryUuid })
-    }
+    const inCategoriesPath = Boolean(matchPath(pathname, `${appModuleUri(designerModules.category)}:uuid/`))
 
-    const stateUpdated = State.assocCategory({ category })(state)
+    const stateUpdated = State.create({ category, categoryUuid, inCategoriesPath })
```
Comment (SteRiccio) [inline webapp/components/survey/CategoryDetails/store/actions/category/useInit.js:24]: I don't think we need the categoryUuid in the state anymore. Once fetched in init, the categoryUuid will be the category.uuid

File: webapp/components/survey/CategoryDetails/store/actions/category/useInit.js [hunk 2]
```diff
@@ -1,54 +1,28 @@
...
-    let category = null
+  return useCallback(async ({ categoryUuid }) => {
+    const category = await API.fetchCategory({ surveyId, categoryUuid })
 
-    if (A.isEmpty(categoryUuid)) {
-      category = await _createCategory({ surveyId, onCategoryCreated, dispatch })
-    } else {
-      category = await _fetchCategory({ surveyId, categoryUuid })
-    }
+    const inCategoriesPath = Boolean(matchPath(pathname, `${appModuleUri(designerModules.category)}:uuid/`))
```
Comment (minotogna) [inline webapp/components/survey/CategoryDetails/store/actions/category/useInit.js:22]: this should not be here. it should belong to init only

File: webapp/components/survey/CategorySelector/CategorySelector.js [hunk 3]
```diff
@@ -73,7 +62,7 @@ const CategorySelector = (props) => {
           {i18n.t('common.manage')}
         </button>
       )}
-      {showCategoriesPanel && !showCategoryPanel && (
+      {showCategoriesPanel && (
```
Comment (minotogna) [inline webapp/components/survey/CategorySelector/CategorySelector.js:65]: does the category list update the list after closing the category panel ?

Comment (ramirobg94) [reply webapp/components/survey/CategorySelector/CategorySelector.js:65]: yes!

File: webapp/components/survey/CategoryDetails/store/actions/category/useInit.js (hunk 2 above)
Comment (ramirobg94) [reply webapp/components/survey/CategoryDetails/store/actions/category/useInit.js:22]: i added an util to put this kind of checks there, idk if it is the right place or put it into some service could be better

File: webapp/utils/isInPath.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+export const useIsInPath = (paths) => {
+  const { pathname } = useLocation()
+  return (paths.constructor === Array ? paths : [paths]).some((path) => Boolean(matchPath(pathname, path)))
+}
+
+export const useInCategoriesPath = () =>
+  useIsInPath([`${appModuleUri(designerModules.category)}:uuid/`, appModuleUri(designerModules.categories)])
+
+export const useInTaxonomiesPath = () =>
+  useIsInPath([`${appModuleUri(designerModules.taxonomy)}:uuid/`, appModuleUri(designerModules.taxonomies)])
```
Comment (minotogna) [inline webapp/utils/isInPath.js:13]: good they are in a common place! but since these are hooks, I'd move them components/hooks

Comment (minotogna) [reply webapp/utils/isInPath.js:13]: call them ```useIsCategoriesRoute```

File: webapp/utils/isInPath.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { matchPath, useLocation } from 'react-router'
+import { designerModules, appModuleUri } from '@webapp/app/appModules'
+
+export const useIsInPath = (paths) => {
+  const { pathname } = useLocation()
+  return (paths.constructor === Array ? paths : [paths]).some((path) => Boolean(matchPath(pathname, path)))
```
Comment (minotogna) [inline webapp/utils/isInPath.js:6]: you can use ```matchPath(pathname, {path})``` where path (which is the argument) can be a string or array

File: webapp/components/survey/NodeDefDetails/store/useNodeDefDetails.js [hunk 6]
```diff
@@ -23,7 +24,7 @@ export const useNodeDefDetails = () => {
 
   const Actions = useActions({ setState })
 
-  const editingFromDesigner = Boolean(matchPath(pathname, `${appModuleUri(designerModules.nodeDef)}:nodeDefUuid`))
+  const editingFromDesigner = useIsInPath(`${appModuleUri(designerModules.nodeDef)}:nodeDefUuid`)
```
Comment (minotogna) [inline webapp/components/survey/NodeDefDetails/store/useNodeDefDetails.js:27]: add the method to common isRoute hook ```useIsDesignerNodeDefRoute```

File: webapp/components/hooks/useIsInRoute.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { matchPath, useLocation } from 'react-router'
+import { designerModules, appModuleUri } from '@webapp/app/appModules'
+
+export const useIsInRoute = (path) => {
+  const { pathname } = useLocation()
+  return matchPath(pathname, path)
```
Comment (minotogna) [inline webapp/components/hooks/useIsInRoute.js:6]: does this work? I see https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/matchPath.md the second argument is an object. 
I thought this should be ```return matchPath(pathname, {path})```

**Pull request merged**